arrow/arrow.py

```diff
@@ -165,46 +189,54 @@ def strptime(cls, date_str, fmt, tzinfo=None):
 
     @classmethod
     def range(cls, frame, start, end=None, tz=None, limit=None):
-        ''' Returns an array of :class:`Arrow <arrow.arrow.Arrow>` objects, representing
+        ''' Returns a list of :class:`Arrow <arrow.arrow.Arrow>` objects, representing
         an iteration of time between two inputs.
 
         :param frame: the timeframe.  Can be any ``datetime`` property (day, hour, minute...).
         :param start: A datetime expression, the start of the range.
         :param end: (optional) A datetime expression, the end of the range.
-        :param tz: (optional) A timezone expression.  Defaults to UTC.
+        :param tz: (optional) A :ref:`timezone expression <tz-expr>`.  Defaults to
+            ``start``'s timezone, or UTC if ``start`` is naive.
         :param limit: (optional) A maximum number of tuples to return.
 
-        **NOTE**: the **end** or **limit** must be provided.  Call with **end** alone to
-        return the entire range, with **limit** alone to return a maximum # of results from the
-        start, and with both to cap a range at a maximum # of results.
+        **NOTE**: The ``end`` or ``limit`` must be provided.  Call with ``end`` alone to
+        return the entire range.  Call with ``limit`` alone to return a maximum # of results from
+        the start.  Call with both to cap a range at a maximum # of results.
+
+        **NOTE**: ``tz`` internally **replaces** the timezones of both ``start`` and ``end`` before
+        iterating.  As such, either call with naive objects and ``tz``, or aware objects from the
+        same timezone and no ``tz``.
 
-        Supported frame values: year, quarter, month, week, day, hour, minute, second
+        Supported frame values: year, quarter, month, week, day, hour, minute, second.
 
         Recognized datetime expressions:
 
             - An :class:`Arrow <arrow.arrow.Arrow>` object.
             - A ``datetime`` object.
 
-        Recognized timezone expressions:
-
-            - A ``tzinfo`` object.
-            - A ``str`` describing a timezone, similar to 'US/Pacific', or 'Europe/Berlin'.
-            - A ``str`` in ISO-8601 style, as in '+07:00'.
-            - A ``str``, one of the following:  'local', 'utc', 'UTC'.
```

Comment:
Rather than duplicating this list everywhere, it's now linked to via `:ref:`timezone expression <tz-expr>``.